### PR TITLE
Fix undocumented golint errors

### PR DESCRIPTION
**Pull request by miguelbernadi**
**What this PR does / why we need it**:
Fixes some golint errors due to missing comment descriptions. These were reported by running `hack/verify-golint.sh` directly, as they were not present in `hack/.golint_failures`.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Related to #68026 

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by miguelbernadi** [comment]
It seems this PR interferes with the code generation. Any idea why? Is it possible to fix or should this just be discarded, as the golint violation is about comments? @dims

**Comment by krmayankk** [comment]
forgot about this,  if this is generated things, then we will have to change this somewhere else . @kubernetes/api-reviewers  might have some hints

**Comment by logicalhan** [comment]
I may be mistaken, but I believe that these files are actually code-generated, which means any lint errors you may be encountering will be reintroduced if these files are regenerated. 

It is probably desirable to fix string which is used to generate this code (i.e. `kubernetes/staging/src/k8s.io/code-generator/cmd/informer-gen/generators/factoryinterface.go`) and then regenerate the files which have this lint error (I think there are more than just these two).

**Comment by logicalhan** [comment]
> It seems this PR interferes with the code generation. Any idea why? Is it possible to fix or should this just be discarded, as the golint violation is about comments? Davanum Srinivas

Oh, I just saw your comment. Yes, I believe that these files are code-generated, so it is probably prudent to resolve this at the source and re-generate the files in order to fix the lint error.

**Comment by logicalhan** [comment]
I've looked into this and it seems that these files should be automatically excluded from lint since they are flagged as being code-generated. Would mind running `golint` against these files directly (without your changes) and letting me know what your output is?

**Comment by lavalamp** [comment]
Please fix at the source: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/code-generator/cmd/informer-gen/generators/factoryinterface.go#L79

**Comment by miguelbernadi** [comment]
> I've looked into this and it seems that these files should be automatically excluded from lint since they are flagged as being code-generated. Would mind running `golint` against these files directly (without your changes) and letting me know what your output is?

```
$ golint ./pkg/client/informers/informers_generated/internalversion/internalinterfaces/factory_interfaces.go
./pkg/client/informers/informers_generated/internalversion/internalinterfaces/factory_interfaces.go:30:6: exported type NewInformerFunc should have comment or be unexported
./pkg/client/informers/informers_generated/internalversion/internalinterfaces/factory_interfaces.go:38:6: exported type TweakListOptionsFunc should have comment or be unexported
```
```
$ golint ./staging/src/k8s.io/sample-controller/pkg/client/informers/externalversions/internalinterfaces/factory_interfaces.go
./staging/src/k8s.io/sample-controller/pkg/client/informers/externalversions/internalinterfaces/factory_interfaces.go:30:6: exported type NewInformerFunc should have comment or be unexported
./staging/src/k8s.io/sample-controller/pkg/client/informers/externalversions/internalinterfaces/factory_interfaces.go:38:6: exported type TweakListOptionsFunc should have comment or be unexported
```

I found them initially by running `hack/verify-golint.sh` before picking a violation to fix, expecting the results of execution would be empty, and found there were several violations.

**Comment by miguelbernadi** [comment]
> Please fix at the source: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/code-generator/cmd/informer-gen/generators/factoryinterface.go#L79

I'll be doing this now. I hope it can fit in this PR in a clear history way, otherwise I'll close this and open a new one for that.

**Comment by miguelbernadi** [comment]
@lavalamp I did change the generator and in a separate commit added the newly generated code. I'm not sure if it should be submitted this way.

File: staging/src/k8s.io/code-generator/cmd/informer-gen/generators/factoryinterface.go [hunk 1]
```diff
@@ -76,6 +76,7 @@ func (g *factoryInterfaceGenerator) GenerateType(c *generator.Context, t *types.
 }
 
 var externalSharedInformerFactoryInterface = `
+// NewInformerFunc is a function returning a SharedIndexInformer
```
Comment (lavalamp) [inline staging/src/k8s.io/code-generator/cmd/informer-gen/generators/factoryinterface.go]: Nit: probably not worth changing, but next time think about customizing the comment for the type. (and ending with a period ;) )

Comment (miguelbernadi) [reply staging/src/k8s.io/code-generator/cmd/informer-gen/generators/factoryinterface.go]: codegen has to be updated for tests to pass, so I'll solve the nit as well. Should I rebase or add them as new commits?

**Comment by miguelbernadi** [comment]
Addressed the nits and rebased on master (was failing verification due to new code-generated files). Added them as new commits, but open to squash them in 2 or 1 commit, as needed.

**Comment by miguelbernadi** [comment]
@lavalamp ☝️

**Comment by mikedanese** [comment]
If we are fixing godoc, can we be sure to remove a package from:

https://github.com/kubernetes/kubernetes/blob/master/hack/.golint_failures

So we don't regress?

File: pkg/client/informers/informers_generated/internalversion/internalinterfaces/factory_interfaces.go [hunk 2]
```diff
@@ -35,4 +36,5 @@ type SharedInformerFactory interface {
 	InformerFor(obj runtime.Object, newFunc NewInformerFunc) cache.SharedIndexInformer
 }
 
+// TweakListOptionsFunc is a function taking v1.ListOptions.
```
Comment (mikedanese) [inline pkg/client/informers/informers_generated/internalversion/internalinterfaces/factory_interfaces.go]: This is pretty sterile. How about "TweakListOptions is a function that transforma a v1.ListOptions"?

Comment (miguelbernadi) [reply pkg/client/informers/informers_generated/internalversion/internalinterfaces/factory_interfaces.go]: I like it. Should I squash some of the previous commits or add a new commit with this change?

Comment (mikedanese) [reply pkg/client/informers/informers_generated/internalversion/internalinterfaces/factory_interfaces.go]: Please squash.

Comment (miguelbernadi) [reply pkg/client/informers/informers_generated/internalversion/internalinterfaces/factory_interfaces.go]: Changed and squashed.

**Comment by miguelbernadi** [comment]
@mikedanese @lavalamp Nits addressed and commits squashed.

File: staging/src/k8s.io/apiextensions-apiserver/examples/client-go/pkg/client/informers/externalversions/internalinterfaces/factory_interfaces.go [hunk 3]
```diff
@@ -35,4 +36,5 @@ type SharedInformerFactory interface {
 	InformerFor(obj runtime.Object, newFunc NewInformerFunc) cache.SharedIndexInformer
 }
 
+// TweakListOptionsFunc is a function that applies a transform on v1.ListOptions.
```
Comment (mikedanese) [inline staging/src/k8s.io/apiextensions-apiserver/examples/client-go/pkg/client/informers/externalversions/internalinterfaces/factory_interfaces.go]: Gah I wrote my comment late at night. "is a function that applies a tranformation to a"? "is a function that transforms a"?

**Pull request merged**